sdk/servicebus/Microsoft.Azure.ServiceBus/src/Amqp/AmqpLinkCreator.cs

```diff
@@ -86,7 +86,7 @@ protected AmqpLinkCreator(string entityPath, ServiceBusConnection serviceBusConn
                     exception);
 
                 session.SafeClose(exception);
-                throw AmqpExceptionHelper.GetClientException(exception, null, link?.GetInnerException(), session.IsClosing());
+                throw AmqpExceptionHelper.GetClientException(exception, null, link?.GetInnerException(), amqpConnection.IsClosing());
```

Comment:
It may be good to comment on this change; without digging into the context, it's not clear why we would use the connection over the session or vice-versa here.

Reply:
Essentially, the name of last arg is `isConnectionError`. We need to be able to distinguish between a regular error and a communication failure. We use `connection.IsClosing` flag to evaluate this value. When service returns an error, it generally returns only at link level (or at times, closes the entire link with the error). Closing of session most of the times is client side activity. For communication issues, connection is closed first, which eventually closed all the sessions and links. So using `session.IsClosing` does work for communication errors as well since if not for client, only communication exception generally leads to session closing.
But in this code, on link open exception, we also have a silly bug where we first trigger `session.SafeClose()` and then check if `session.IsClosing()`, which as Sean mentioned is always true.

Reply:
Thank you for the expanded context.  This makes more sense to me now.   It may be worth putting just a short mention in a code comment for those wondering.

Reply:
I did think of adding a comment, but can hardly come up with a comment which describes why I am using `amqpConnection.IsClosing()` for the argument `isConnectionError`.
The person who goes through the PR might have a confusion as to what's happening, but just looking at the current state of the code, it might not be a confusing thing.